aineko/models/config_schema.py

```diff
@@ -16,16 +16,16 @@ class Dataset(BaseModel):
             """Dataset model."""
 
             type: str
-            params: Optional[dict]
+            params: Optional[dict] = None
 
         class Node(BaseModel):
             """Node model."""
 
             class_name: str = Field(..., alias="class")
-            node_params: Optional[dict]
-            node_settings: Optional[dict]
-            inputs: Optional[list]
-            outputs: Optional[list]
+            node_params: Optional[dict] = None
+            node_settings: Optional[dict] = None
+            inputs: Optional[list] = None
+            outputs: Optional[list] = None
```

Comment:
Trying to understand this change. What is the rational for adding defaults to None throughout? Isn't this a change in behavior? I thought that `Optional[str]` by itself did not specifically have a default. I didn't fully trace the usage, do we use a default anywhere?

Reply:
`Optional[str]` by itself is just a type hint and shouldn't actually set None as a default value. But it's recommended to add the default value: [Required, optional, and nullable fields](https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields)

Reply:
Ok understood